lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2399,10 +2408,28 @@ LogicalResult StmtEmitter::visitStmt(InstanceOp op) {
 
     // Emit the value as an expression.
     ops.clear();
-    // output ports were lowered to wire
-    if (elt.isOutput())
+    // output ports that are not connected to single use output ports were lowered to wire
+    OutputOp output = dyn_cast_or_null<OutputOp>(portVal.getUses().begin()->getOwner());
```

Comment:
I don't think you always have even one use.

Reply:
I think the `portVal` has at least one use, which is assigning it to the output port of the parent module.

Reply:
I don't think this is correct for non-output ports, which can have zero users.